assembl/static/js/app/views/message.js

```diff
@@ -1524,6 +1546,52 @@ var MessageView = Marionette.LayoutView.extend({
     this.listenToOnce(this.messageModerationOptionsView, 'moderationOptionsClose', this.onModerationOptionsClose);
   },
 
+  onDeleteMessageClick: function(ev){
+    var that = this;
+    // We could try to minimize context switching for the user, by scrolling the viewport to the message the user wants to delete, as soon as the confirmation popin opens, using this line of code:
+    // that.messageListView.scrollToMessage(that.model, false, false);
+    
+    var onSubmit = function(ev){
+      var analytics = Analytics.getInstance();
+      analytics.trackEvent(analytics.events.MESSAGE_LIKED);
+      // we could use that.model.destroy() instead, and add || method == "delete" to models/message.js::sync()
+      var message_delete_url = that.model.getApiV2Url();
+      Promise.resolve(
+        $.ajax(message_delete_url, {
+          method: "DELETE",
+          contentType: "application/json",
+          dataType: "json"
+        })
+      ).then(function(data) {
+        Growl.showBottomGrowl(
+          Growl.GrowlReason.SUCCESS,
+          i18n.gettext('Message has been successfully deleted.'),
+          { delay: 12000 }
+        );
+
+        // Refresh the messageList
+        that.messageListView.render();
+        setTimeout(function(){
+          that.messageListView.showMessageById(that.model.id, null, true, false);
+        }, 500);
+        
+      }).catch(function(e) {
+        Growl.showBottomGrowl(
+          Growl.GrowlReason.ERROR,
+          i18n.gettext('Error: Message could not be deleted.'),
+          { delay: 12000 }
+        );
+      });
+    };
+    var confirm = new ConfirmModal({
+      contentText: i18n.gettext('Are you sure you want to delete this message?'),
+      cancelText: i18n.gettext('No'),
+      submitText: i18n.gettext('Yes'),
+      onSubmit: onSubmit,
+    });
+    Assembl.slider.show(confirm);
```

Comment:
Should we use the `@app.common.context.setCurrentModalView` as well? Will the confirmation box be closed by any other view other than itself? If not, then no worries.